src/grpc/protocol/keyprovider_structs.rs

```diff
@@ -3,11 +3,24 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use anyhow::{anyhow, Result};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
+use std::convert::TryFrom;
 use std::vec::Vec;
 
-#[derive(Serialize, Deserialize, Debug)]
+pub const ERR_INVALID_OP: &str = "invalid operator";
+pub const ERR_MISSING_OP: &str = "missing operator";
+pub const ERR_UNSUPPORTED_OP: &str = "unsupported operator";
+pub const ERR_UNWRAP_PARAMS_NO_ANNOTATION: &str = "keyunwrap parameters must include annotation";
+pub const ERR_UNWRAP_PARAMS_NO_DC: &str = "keyunwrap parameters must include dc";
+pub const ERR_WRAP_PARAMS_EXPECT_EMPTY_EC: &str = "keywrap parameters should not include ec";
+pub const ERR_WRAP_PARAMS_EXPECT_EMPTY_OPTSDATA: &str =
+    "keywrap parameters should not include optsdata";
```

Comment:
The Attestation Agent doesn't wrap keys, hence the comments in the original that say that the ec and opts should be null. I'm not sure if this is a requirement that we need to strictly enforce, however. A keywrap request without an encryption config is in fact not a valid keywrap request. Keywrap parameters should include and ec and optsdata, you just shouldn't ever be dealing with them in the AA. 

We have talked about trying to reuse some code between the AA and the sample KBS (which does wrap keys). These plans are tentative and not a huge priority, but would require reworking this to support (and require) keywrap requests with the necessary fields.

Reply:
@fitzthum - ok. I'm not totally clear from your comment whether you'd like me to remove that additional checking?

Reply:
I think it's technically fine as it is. I guess we could skip the valid check on the keywrapparams. We don't really need some of the related functionality or tests but it doesn't hurt.